code/game/objects/items/implants/implant_exile.dm

```diff
@@ -20,3 +20,44 @@
 	name = "implant case - 'Exile'"
 	desc = "A glass case containing an exile implant."
 	imp_type = /obj/item/implant/exile
+
+
+//For hotel staff, prevents them entering the station Z-level, dusts you if you remove it
+
+/obj/item/implant/exile/hotel
+	name = "super exile implant"
+	desc = "nice try hotel staff"
+
+/obj/item/implant/exile/hotel/on_implanted(mob/user)
+	user.AddComponent(/datum/component/stationloving/hotelloving)
+
+/obj/item/implant/exile/hotel/removed(mob/source) // Incase they try self surgery
+	visible_message("<span class='danger'>The implant's anti-removal mechanisms activate!")
+	source.dust()
+	message_admins("[ADMIN_LOOKUPFLW(source)] tried to remove their hotel staff implant to enter the station and was dusted.")
+	if(!QDELETED(src)) //If you try to qdel when the implant is removed without an implant case it causes a loop of qdels and gibbing
+		qdel(src)
+
+/datum/component/stationloving/hotelloving/in_bounds()
+	if(SSticker.current_state <= GAME_STATE_PREGAME)
+		return TRUE
+	var/turf/T = get_turf(parent)
+	if (!T)
+		return FALSE
+	if (is_station_level(T.z)) // Are they on the station Z-level? If so trigger relocate()
+		return FALSE
+	return TRUE
+
+//Override to plop the disk back to a syndie crew spawn rather than somewhere on the station.
+
+/datum/component/stationloving/hotelloving/relocate()
+	var/mob/AM = parent
+	if(ismob(AM))
+		if(!QDELETED(src)) // if you don't do this the body gets continuously dusted forever. While this is funny, an infinitely large pile of remains that crashes clients on right click isn't.
+			qdel(src)
+		to_chat(AM,"<span class='danger'>The implant's anti-escape mechanisms activate!")
+		AM.dust() // Nice try hotel staff
+		message_admins("[ADMIN_LOOKUPFLW(source)] tried to enter the station as hotel staff and was dusted.")
```

Comment:
I do not see a source var defined anywhere

Reply:
daannng just beat me to this

Reply:
ahem typo. Oops.